frame/staking/src/pallet/impls.rs

```diff
@@ -893,12 +904,13 @@ impl<T: Config> Pallet<T> {
 	/// `Nominators` or `VoterList` outside of this function is almost certainly
 	/// wrong.
 	pub fn do_remove_nominator(who: &T::AccountId) -> bool {
-		let outcome = if Nominators::<T>::contains_key(who) {
+		let outcome = if let Some(nominations) = Self::nominations(who) {
 			Nominators::<T>::remove(who);
 			let _ = T::VoterList::on_remove(who).defensive();
+			let _ = T::EventListener::on_nominator_remove(who, nominations);
 			true
 		} else {
-			false
+			true
 		};
 
 		debug_assert_eq!(
```

Comment:
This should go into `try_state` if it does not already exist.

Reply:
Already done yeah, I'll remove this